lib/http/response/body.rb

```diff
@@ -53,7 +53,10 @@ def to_s
           @streaming  = false
           @contents   = String.new("").force_encoding(encoding)
 
-          while (chunk = @stream.readpartial)
+          length = @length || Float::INFINITY
```

Comment:
I think we should follow your initial idea of limiting amount of data to given Content-Length when available and should be respected according to RFC.